sdk/confidentialledger/Azure.Storage.ConfidentialLedger/src/Generated/ConfidentialLedgerClient.cs

```diff
@@ -46,29 +44,94 @@ public ConfidentialLedgerClient(Uri ledgerUri, TokenCredential credential, Confi
             }
 
             options ??= new ConfidentialLedgerClientOptions();
-            Pipeline = HttpPipelineBuilder.Build(options, new BearerTokenAuthenticationPolicy(credential, AuthorizationScopes));
+            _clientDiagnostics = new ClientDiagnostics(options);
+            var authPolicy = new BearerTokenAuthenticationPolicy(credential, AuthorizationScopes);
+            Pipeline = HttpPipelineBuilder.Build(options, new HttpPipelinePolicy[] { authPolicy, new LowLevelCallbackPolicy() });
             this.ledgerUri = ledgerUri;
             apiVersion = options.Version;
         }
 
         /// <summary> The constitution is a script that assesses and applies proposals from consortium members. </summary>
-        /// <param name="cancellationToken"> The cancellation token to use. </param>
-        public virtual async Task<Response> GetConstitutionAsync(CancellationToken cancellationToken = default)
+        /// <param name="requestOptions"> The request options. </param>
+#pragma warning disable AZC0002
+        public virtual async Task<Response> GetConstitutionAsync(RequestOptions requestOptions = null)
+#pragma warning restore AZC0002
         {
-            Request req = CreateGetConstitutionRequest();
-            return await Pipeline.SendRequestAsync(req, cancellationToken).ConfigureAwait(false);
+            requestOptions ??= new RequestOptions();
+            HttpMessage message = CreateGetConstitutionRequest(requestOptions);
+            if (requestOptions.PerCallPolicy != null)
+            {
+                message.SetProperty("RequestOptionsPerCallPolicyCallback", requestOptions.PerCallPolicy);
```

Comment:
This also worries me. It forces us to allocate a hashtable per call. We should measure the impact and either optimize or simplify this feature, i.e. I don't think we should make the whole pipeline less efficient to support this corner case feature.

Reply:
Same as above, once we know for certain we want this we can move it to Azure.Core and not punch it in so strongly.

This will only hit LLC usages right now.